chia/daemon/server.py

```diff
@@ -198,52 +205,48 @@ def cancel_task_safe(self, task: Optional[asyncio.Task]):
 
     async def stop(self) -> Dict[str, Any]:
         self.shut_down = True
-        self.cancel_task_safe(self.ping_job)
         await self.exit()
         if self.websocket_server is not None:
             self.websocket_server.close()
+        self.shutdownd_event.set()
         return {"success": True}
 
-    async def safe_handle(self, websocket: WebSocketServerProtocol, path: str):
-        service_name = ""
-        try:
-            async for message in websocket:
-                try:
-                    decoded = json.loads(message)
-                    if "data" not in decoded:
-                        decoded["data"] = {}
-                    response, sockets_to_use = await self.handle_message(websocket, decoded)
-                except Exception as e:
-                    tb = traceback.format_exc()
-                    self.log.error(f"Error while handling message: {tb}")
-                    error = {"success": False, "error": f"{e}"}
-                    response = format_response(decoded, error)
-                    sockets_to_use = []
-                if len(sockets_to_use) > 0:
-                    for socket in sockets_to_use:
-                        try:
-                            await socket.send(response)
-                        except Exception as e:
-                            tb = traceback.format_exc()
-                            self.log.error(f"Unexpected exception trying to send to websocket: {e} {tb}")
-                            self.remove_connection(socket)
-                            await socket.close()
-        except Exception as e:
-            tb = traceback.format_exc()
-            service_name = "Unknown"
-            if websocket in self.remote_address_map:
-                service_name = self.remote_address_map[websocket]
-            if isinstance(e, ConnectionClosedOK):
-                self.log.info(f"ConnectionClosedOk. Closing websocket with {service_name} {e}")
-            elif isinstance(e, WebSocketException):
-                self.log.info(f"Websocket exception. Closing websocket with {service_name} {e} {tb}")
-            else:
-                self.log.error(f"Unexpected exception in websocket: {e} {tb}")
-        finally:
-            self.remove_connection(websocket)
-            await websocket.close()
+    async def incoming_connection(self, request):
+        ws: WebSocketResponse = web.WebSocketResponse(max_msg_size=self.daemon_max_message_size, heartbeat=30)
```

Comment:
Does this `heartbeat=30` override the `heartbeat=60` specified in either "connect_to_daemon" or the `daemon/client.py`

Reply:
This is replacing the `ping` task that ws being manually executed every 30 seconds, do it does not change the timeout from the daemon side.
The RPC service side is still at 60. I suppose it's fine to have different ping times for each side of the socket.

Reply:
Maybe we should hardcode these constants at the top of the file as well (heartbeat, keepalive timeout etc)

Reply:
Oh right, I didn't notice that ping task - nice to get rid of that.